### ISSUE TITLE
Google Guava support

### ISSUE THREAD

**Issue by tomazic89** [javers/javers#80]
This is more an 'enhancement' than a bug I suppose.
javers.compare() doesn't detect changes in the object variables of type com.google.common.collectMultiMap.

Since Guava is increasingly used this feature would be nice to have.

Is it possible to register my own comparator for custom types? Can't find that in the documentation.

Thanks, Nejc

**Comment by tomazic89** [comment]
I respect lightweight decision and I'd also like to leave it like that.
I currently don't have a case that I would need Multimap attributes to be compared, but I believe that I will  in the near future.
MapChange seems like a good class to start with and I believe that it would solve my case. 
I can give it a closer look and maybe propose an API in 2 days, I'm totally overbooked till Sunday.

In my mind I was thinking of an API that would allow registering a Class<T> type and a comparator for it which would be called wen such a class is encountered (maybe also expected ChangeType out of existing ones). That kind of API would allow users to simply generate custom diffs or even generating changes for unsupported types (Like Guava MultiMap). I'm not sure if this is possible right away without any extra dependencies, so as I said, will give it a closer look in a couple of days. 
If get any ideas, I would be glad to test them.

KR, nejc

**Comment by bartoszwalacik** [comment]
ok, I'm starting to work on this issue, soon, I'll give you an insight into new API

**Comment by bartoszwalacik** [comment]
I have came up with this simple interface
```
/**
 * Custom Map comparator.
 * <br/>
 * Useful for classes that are logically maps but do not implement java.util.Map interface.
 * <br/><br/>
 *
 * Implementation should calculate diff between two maps of type T
 * and return the result as MapChange.
 * For example, T could be a Multimap from Guava.
 */
public interface CustomMapComparator<T> extends CustomPropertyComparator<T, MapChange>{

    MapChange compare(T left, T right);
}
```

**Comment by pawelszymczyk** [comment]
I have noticed that Jackson had resolved similiar problem:
https://github.com/FasterXML/jackson-databind/blob/master/src/test/java/com/fasterxml/jackson/databind/module/TestTypeModifiers.java#L218

The had MyMapLikeType and posibility to register custom modifiers

**Comment by bartoszwalacik** [comment]
done and released in 1.0.4, there is no documentation fot this feature yet, 
try 
http://javers.org/javadoc_1.0.0/org/javers/core/JaversBuilder.html#registerCustomComparator-org.javers.core.diff.custom.CustomPropertyComparator-java.lang.Class-

**Comment by bartoszwalacik** [comment]
give us a star if you like javers
https://github.com/javers/javers/stargazers
